### PR TITLE
Solves pull/145

### PR DESCRIPTION
**Pull request by odahcam**
https://github.com/danielstjules/Stringy/pull/145/

### CODE REVIEW

File: src/Stringy.php [hunk 1]
```diff
@@ -463,8 +463,21 @@ public function hasUpperCase()
      */
     public function htmlDecode($flags = ENT_COMPAT)
     {
-        $str = html_entity_decode($this->str, $flags, $this->encoding);
-
+        if (function_exists('html_entity_decode')) {
```
Comment (danielstjules) [inline src/Stringy.php]: I still don't understand, when and why would `html_entity_decode` not be used? It's part of the standard library.

File: src/Stringy.php [hunk 2]
```diff
@@ -463,8 +463,21 @@ public function hasUpperCase()
...
+            $str = html_entity_decode($this->str, $flags, $this->encoding);
+        } else {
+            // replace numeric entities
```
Comment (danielstjules) [inline src/Stringy.php]: Will this branch ever run?

**Comment by danielstjules** [comment]
Appreciate the PR! But I think we'll stick to the current method logic. Thanks!

**Comment by odahcam** [comment]
Oh! Look at me trying to support `PHP@'<=4.3.0'` 😆 

Thanks for not accepting that.

**Pull request closed without merging**